comictaggerlib/taggerwindow.py

```diff
@@ -1508,20 +1552,23 @@ def remove_tags(self, style: str) -> None:
                     progdialog.setValue(prog_idx)
                     progdialog.setLabelText(str(ca.path))
                     QtCore.QCoreApplication.processEvents()
-                    if ca.has_metadata(style) and ca.is_writable():
-                        if not ca.remove_metadata(style):
-                            failed_list.append(ca.path)
-                        else:
-                            success_count += 1
-                        ca.load_cache(list(metadata_styles))
+                    for style in styles:
+                        if ca.has_metadata(style) and ca.is_writable():
+                            if not ca.remove_metadata(style):
+                                failed_list.append(ca.path)
+                                # Abandon any further tag removals to prevent any greater damage to archive
+                                break
+                            else:
+                                success_count += 1
+                            ca.load_cache(list(metadata_styles))
```

Comment:
lets also reset the cache before loading the cache

Reply:
Should the `reset_cache` and `load_cache` be outside the loop?

Reply:
Outside of the style loop